### PR TITLE
Reworks Limb Gibbing Mechanics

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Limb gibbing mechanics on Aurora were pretty bad. For melee weapons, it was RNG based and somewhat fair because it was actually based on mechanics. For bullets, it used custom snowflake code that was just absolutely insulting. It didn't really regard armor like most servers, and unlike most servers, if your feet are gibbed you're sideways until it's fixed.

This rework makes it so it's less reliant on RNG and more fair. RNG was completely removed when calculating limb gibbing, however to prevent gibbing on the first shot, each limb cannot be gibbed on the first shot. Basically if you shoot a bullet, and it was meant to gib, it won't gib but instead set a flag that tells it to gib the next time it's supposed to gib. This should actually make it so combat doesn't consist of aiming at the feet with the .367 or head gibbing with shotgun slugs. Also, to prevent further memes, armor now actually matters when getting hit by a bullet. Before, armor mattered very little and marginally affected limb gibbing chances.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
I tested it and I will likely remove the arbitrary "never gib on the first hit" variable as I accidentally coded something less terrible than that.

**Comment by comma** [comment]
>It didn't really regard armor

Actually by the time you reach organ's damage proc it's assumed you adjusted all shit with armor/shields/etc, in mob-level procs. The impression that armor does jack shit is probably from there actually not being any armor due to armor vests not covering limbs and pretty much no items giving non-joke foot armor.

**Comment by BurgerLUA** [comment]
!review

**Comment by BurgerLUA** [comment]
!review

File: html/changelogs/burgerbb - maim tweak.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - balance: "Limbs cannot be gibbed if the limb's armor value exceeds the damage dealt. Limbs cannot also be gibbed on the first hit. Limb gibbing is no longer determined by RNG."
```
Comment (MarinaGryphon) [inline html/changelogs/burgerbb - maim tweak.yml]: The "first hit" variable was removed, right?

Comment (BurgerLUA) [reply html/changelogs/burgerbb - maim tweak.yml]: Yeah

**Comment by skull132** [comment]
This is now test-merged live.

**Comment by skull132** [comment]
This was test-merged for a day, and no issues were put up. So, looks good to roll and I'm going to shove it in.

**Comment by BurgerLUA** [comment]
That is what she said.

**Pull request merged**